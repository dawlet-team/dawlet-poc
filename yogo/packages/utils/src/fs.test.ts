import {
  fetchAvailableDawlets,
  readDawletsDir,
  getPackageJson,
  retrieveDawletConfig,
  pathToDawletsDir,
  pkgNameWithoutPrefix
} from "./fs";

describe("path to dawlets folder", () => {
  it("contains an absolute path", () => {
    expect(pathToDawletsDir).toMatch(/\/dawlet-poc\/yogo\/dawlets$/);
  });
});

describe("pkgNameWithoutPrefix", () => {
  it("throws error if the package is not a dawlet package", () => {
    expect(() => {
      pkgNameWithoutPrefix("example-let");
    }).toThrowErrorMatchingInlineSnapshot(
      `"example-let is not a Dawlet package"`
    );
  });
  it("returns pkgName without @dawlet", () => {
    expect(pkgNameWithoutPrefix("@dawlet/algolet")).toEqual("algolet")
  })
});

describe("fetchAvailableDawlets", () => {
  it("reads dawlets folder", () => {
    const dawlets = readDawletsDir();
    expect(dawlets).toMatchSnapshot();
  });
  it("fetches available dawlets", () => {
    expect(fetchAvailableDawlets()).toMatchSnapshot();
  });
});

describe("getPackageJson", () => {
  it("gets package json", () => {
    const json = getPackageJson("algolet");
    expect(json).toBeInstanceOf(Object);
  });
});

describe("retrieveDawletConfig", () => {
  it("gets dawlet config", () => {
    const algoletConfig = retrieveDawletConfig("algolet");
    expect(algoletConfig).toMatchSnapshot();
  });
});
