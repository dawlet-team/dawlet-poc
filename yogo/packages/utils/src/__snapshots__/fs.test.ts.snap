// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchAvailableDawlets fetches available dawlets 1`] = `
Array [
  Object {
    "description": "Algolet lets you write your own custom logic to generate music.",
    "localizable": true,
    "name": "@dawlet/algolet",
  },
  Object {
    "description": "Sheetlet displays the music represented as a sheetmusic.",
    "localizable": true,
    "name": "@dawlet/sheetlet",
  },
  Object {
    "description": "Synthlet provides a simple synthesizer to playback the music. It has a dedicated local transport on its own. ",
    "localizable": true,
    "name": "@dawlet/synthlet",
  },
]
`;

exports[`fetchAvailableDawlets reads dawlets folder 1`] = `
Array [
  "algolet",
  "sheetlet",
  "synthlet",
]
`;

exports[`retrieveDawletConfig gets dawlet config 1`] = `
Object {
  "description": "Algolet lets you write your own custom logic to generate music.",
  "localizable": true,
  "name": "@dawlet/algolet",
}
`;
